VotingApplication/VotingApplication.Web.Api.Tests/E2E/BasicPollTests.cs

```diff
@@ -0,0 +1,603 @@
+                _inviteOnlyBasicPoll = new Poll()
+                {
+                    UUID = Guid.NewGuid(),
+                    PollType = PollType.Basic,
+                    Name = "Test Poll",
+                    LastUpdated = DateTime.Now,
+                    CreatedDate = DateTime.Now,
+                    Options = testPollOptions,
+                    InviteOnly = true,
+                    NamedVoting = false,
+                    OptionAdding = false,
+                    HiddenResults = false,
+                    Ballots = new List<Ballot>()
+                    {
+                        new Ballot() { TokenGuid = Guid.NewGuid() }
+                    }
+                };
+
+                _context.Polls.Add(_inviteOnlyBasicPoll);
+                _context.SaveChanges();
+            }
+
+            [ClassCleanup]
+            public static void ClassCleanup()
+            {
+                PollClearer pollTearDown = new PollClearer(_context);
+                pollTearDown.ClearPoll(_inviteOnlyBasicPoll);
+
+                _context.Dispose();
+            }
+
+            [TestMethod, TestCategory("E2E")]
+            public void BasicVote_AccessWithNoToken_DisplaysError()
+            {
+                _driver.Navigate().GoToUrl(SiteBaseUri + "Poll/#/Vote/" + _inviteOnlyBasicPoll.UUID);
+                IWebElement error = _driver.FindElement(NgBy.Binding("$root.error.readableMessage"));
+
+                Assert.IsTrue(error.IsVisible());
+                Assert.AreEqual("This poll is invite only", error.Text);
+            }
+
+            [TestMethod, TestCategory("E2E")]
+            public void BasicVote_AccessWithToken_DisplaysAllOptions()
+            {
+                _driver.Navigate().GoToUrl(SiteBaseUri + "Poll/#/Vote/" + _inviteOnlyBasicPoll.UUID + "/" + _inviteOnlyBasicPoll.Ballots[0].TokenGuid);
+                IReadOnlyCollection<IWebElement> optionNames = _driver.FindElements(NgBy.Binding("option.Name"));
+
+                Assert.AreEqual(_inviteOnlyBasicPoll.Options.Count, optionNames.Count);
+                CollectionAssert.AreEquivalent(_inviteOnlyBasicPoll.Options.Select(o => o.Name).ToList(),
+                                               optionNames.Select(o => o.Text).ToList());
+            }
+
+            [TestMethod, TestCategory("E2E")]
+            public void BasicVote_VoteWithToken_NavigatesToResultsPage()
+            {
+                _driver.Navigate().GoToUrl(SiteBaseUri + "Poll/#/Vote/" + _inviteOnlyBasicPoll.UUID + "/" + _inviteOnlyBasicPoll.Ballots[0].TokenGuid);
+                IReadOnlyCollection<IWebElement> buttons = _driver.FindElements(By.TagName("Button"));
+                buttons.First(b => b.Text == "Vote").Click();
+
+                VoteClearer voterClearer = new VoteClearer(_context);
+                voterClearer.ClearLast();
+
+                Assert.IsTrue(_driver.Url.StartsWith(SiteBaseUri + "Poll/#/Results/" + _inviteOnlyBasicPoll.UUID));
```

Comment:
Ahh, that makes sense. I missed that the url has a token on the end. `StartsWith` is fine then.